casper/src/test/scala/coop/rchain/casper/util/rholang/RuntimeManagerTest.scala

```diff
@@ -80,4 +76,36 @@ class RuntimeManagerTest extends FlatSpec with Matchers {
 
     hash1 should be(hash2)
   }
+
+  "computeState" should "charge deploys separately" in {
+    val terms = List(
+      """for(@x <- @"w") { @"z"!("Got x") }""",
+      """for(@x <- @"x"; @y <- @"y"){ @"xy"!(x + y) } | @"x"!(1) | @"y"!(10)"""
+    )
+
+    def deployCost(p: Seq[InternalProcessedDeploy]): Long = p.map(_.cost).map(_.cost).sum
+    val deploy = terms.map(t =>
+      ProtoUtil.termDeploy(InterpreterUtil.mkTerm(t).right.get, System.currentTimeMillis()))
```

Comment:
You should use `ProtoUtil.souceDeploy` instead.

Reply:
Sure.

Reply:
Wait, why is that? Tests pass so what's the reason?

Reply:
Nevermind, my mistake, this is good as is.